### PR TITLE
feat(pill): migrate to V4

### PR DESCRIPTION
**Pull request by massao**
# Purpose of PR

Migrate `Pill` component to v4

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/pill/src/Pill.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    onDrag,
+    className,
+    dragHandleComponent,
+    variant = 'idle',
+    ...otherProps
+  } = props;
+
+  const styles = getPillStyles(variant);
+
+  console.log(onDrag);
```
Comment (thebesson) [inline packages/components/pill/src/Pill.tsx:37]: rogue `console.log`

File: packages/components/pill/src/Pill.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { CommonProps, Box } from '@contentful/f36-core';
+import { TabFocusTrap } from '@contentful/f36-utils';
+import { Drag, Close } from '@contentful/f36-icons';
+import { PillVariants } from './types';
+import getPillStyles from './styles';
+
+export interface PillProps
+  extends React.HTMLAttributes<HTMLElement>,
+    CommonProps {
+  label: string;
```
Comment (suevalov) [inline packages/components/pill/src/Pill.tsx:12]: Ideally, we would have jsdoc comments for all 5 properties here, because there are picked up by tsdoc generator and end up on the documentation page.

File: packages/components/pill/src/Pill.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            <Close className={styles.icon} variant="muted" />
+          </TabFocusTrap>
+        </button>
+      )}
+    </Box>
+  );
+}
+
+/**
+ * TODO: Add description of component here.
```
Comment (suevalov) [inline packages/components/pill/src/Pill.tsx:74]: Don't forget to remove it

File: packages/components/pill/src/Pill.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      onDrag={onDrag}
+      ref={ref}
+      {...otherProps}
+    >
+      {onDrag &&
+        (dragHandleComponent ? (
+          dragHandleComponent
+        ) : (
+          <span className={styles.dragIcon}>
+            <Drag className={styles.icon} variant="muted" />
```
Comment (suevalov) [inline packages/components/pill/src/Pill.tsx:51]: I wonder if we need to apply any aria labels to the drag component here

File: packages/components/pill/src/Pill.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          </span>
+        ))}
+      <span aria-label={label} title={label} className={styles.label}>
```
Comment (suevalov) [inline packages/components/pill/src/Pill.tsx:54]: Not sure if we need `title` and `aria-label` at the same time

**Review by suevalov — COMMENTED** [review]
Don't forget to add automatically generate props table to the documentation page.

Reviewing and modifying the documentation itself won't hurt either.

File: packages/components/pill/src/Pill.tsx (hunk 4 above)
Comment (massao) [reply packages/components/pill/src/Pill.tsx:51]: This would be good, since we don't have any visible indication that this is a drag handle.

File: packages/components/pill/src/Pill.tsx (hunk 5 above)
Comment (massao) [reply packages/components/pill/src/Pill.tsx:54]: Just checked and actually the `aria-label` should only be used when there is no visible text that acts as the label, so will keep only the title for it.

File: packages/components/pill/src/Pill.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { cx } from 'emotion';
+import { CommonProps, Box } from '@contentful/f36-core';
+import { TabFocusTrap } from '@contentful/f36-utils';
+import { Drag, Close } from '@contentful/f36-icons';
+import { PillVariants } from './types';
+import getPillStyles from './styles';
+
+export interface PillProps
+  extends React.HTMLAttributes<HTMLElement>,
```
Comment (suevalov) [inline packages/components/pill/src/Pill.tsx:10]: Do we actually want to allow all `React.HTMLAttributes<HTMLElement>` to be passed or just specific props + CommonProps will be enough?

Comment (suevalov) [reply packages/components/pill/src/Pill.tsx:10]: I started wondering because the props table generated automatically looks huge

**Review by suevalov — COMMENTED** [review]
One last comment about generic HTMLDiv props. Otherwise, looks good.

File: packages/components/pill/README.mdx [hunk 7]
```diff
@@ -60,4 +64,9 @@ Example of the Pill component with dragHandle, in that case Icon.
 
 ## Accessibility
 
-- Label value is passed in the component as a aria-label and title in the span element, like follows `aria-label={label} title={label}`.
+- Label value is passed in the component as a title in the span element, like follows `title={label}`.
+- The default drag component has a `aria-label="Drag handler"`, custom components should also include an `aria-label`
```
Comment (burakukula) [inline packages/components/pill/README.mdx:68]: It is not 100% clear to me what we mean by custom components. I think an example here could be nice.

File: packages/components/pill/src/Pill.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    CommonProps {
+  /**
+   * Text that will be shown on the pill
+   */
+  label: string;
+  /**
+   * Function that handles when the close icon is clicked. Close icon visibility also depends on if this is set
```
Comment (burakukula) [inline packages/components/pill/src/Pill.tsx:17]: Minor thing, I would rephrase here a bit: 
(... ) Close icon visibility depends on if this property is set.
Now when I think about it, it would be good to maybe extend to code styleguide with guidelines around comments. Or it's overkill?

**Comment by burakukula** [comment]
@massao thanks for the contribution! Great work! 💯 
A small visual thing that I noticed is the alignment of the icon inside the Pill. Maybe there could be a way to vertically align it a bit more?

Screenshot 2021-07-08 at 09 25 06

File: packages/components/pill/stories/Pill.stories.tsx [hunk 9]
```diff
@@ -51,7 +62,13 @@ export const CustomHandleComponent = (args: PillProps) => (
...
 );
 
-export const PillVariants = (args: PillProps) => (
+customHandleComponent.args = {
+  label: 'example.user@contentful.com',
+  onClose: action('clicked'),
+  onDrag: action('dragged'),
+};
+
+export const PillVariants: Story<PillProps> = (args: PillProps) => (
```
Comment (burakukula) [inline packages/components/pill/stories/Pill.stories.tsx:71]: A small thing that came to my mind is that maybe we could extend the existing overview story to contain the `deleted` variant with a close icon.

File: packages/components/pill/src/Pill.tsx (hunk 8 above)
Comment (massao) [reply packages/components/pill/src/Pill.tsx:17]: yeah, that phrasing looks better, will update it for now, and we can talk about the guidelines for comments later, to have it consistent across all components.

File: packages/components/pill/src/Pill.tsx (hunk 6 above)
Comment (massao) [reply packages/components/pill/src/Pill.tsx:10]: That's a good question, I kept it with the `React.HTMLAttributes<HTMLElement>` as it was like that on the v3, but we can remove it if it's not necessary.
Or maybe only use the `React.HTMLAttributes<HTMLDivElement>` it should remove some of the props from the generated table.
WDYT??

**Review by burakukula — APPROVED** [review]
👏 Awesome work! Thanks for the contribution!

**Review by suevalov — APPROVED** [review]

**Review by thebesson — APPROVED** [review]
